client/app/bundles/momentDashboards/components/ChartControl.jsx

```diff
@@ -0,0 +1,71 @@
+// @flow
+import React from 'react';
+import _ from 'lodash';
+import Chart from './Chart';
+
+type chartControlState = {
+  type: string;
+  data: any;
+}
+
+type chartControlProp = {
+  types: string[];
+  initialParams: {
+    type: string,
+    data: {}
+  };
+}
+
+const ChartControlButton =
+  (index: number, type: string, onClick: (type: string) => void) => (
+    <button
+      className={'ui-button ui-widget ui-corner-all'}
+      key={index}
+      onClick={onClick}
+    >
+      {type}
+    </button>);
+
+/**
+ * Control Panel for selecting different objects to graph.
+ */
+export default class ChartControl extends React.Component {
+  props: chartControlProp;
+  state: chartControlState;
+
+  constructor(props: chartControlProp) {
+    super(props);
+
+    const { initialParams } = this.props;
+
+    this.state = {
+      type: initialParams.type,
+      data: initialParams.data,
+    };
+  }
+
+  onSelectType(value: string) {
+    return () => {
+      this.setState({type: value});
+    };
+  }
+
+  render() {
+    const {types} = this.props;
+    const buttons: any[] = [];
+    _.each(types, (value, index: number) => {
+      buttons.push(new ChartControlButton(index, value, this.onSelectType(value)));
```

Comment:
This is the first time I've seen React FSC (functional stateless components) instantiated using **new**. Generally I'm more used to seeing:

`const ChartControlButton = ({type, ...props}: ChartControlButtonProps) => (<button {...props}>{type}</button>);`
`buttons.push(<ChartControlButton {...{index, value, onClick: this.onSelectType(value)}}/>`

Reply:
oh good catch, it was originally a class, but it passed lint/@flow without issues. Will definitely check it out on the React tools.

Now I'm also curious to see if the transpiled code passses it in like this as well, or if  it is wrapped in `react.createComponent()`